maniphttp/internal/syscall/syscall_linux.go

```diff
@@ -0,0 +1,43 @@
+// +build linux
+
+package syscall
+
+// package to set to the low-level/OS settings
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	defaultTCPNetworkTimeout = 30 * time.Second
+)
+
+// SetTCPUserTimeout sets the TCP timeout for a socket connection
+func SetTCPUserTimeout(t time.Duration) func(string, string, syscall.RawConn) error {
+
+	// return if the tcpUserTimeout is not set.
+	if t == 0 {
+		return nil
+	}
+	tcpTimeout := defaultTCPNetworkTimeout
```

Comment:
can this whole logic be just tcpTimeout = t ? can t be less than 0?

Reply:
yes make sense, I guess in the 2nd iteration I added that zero check and didn't optimize the below logic. :)